driver.py

```diff
@@ -407,14 +445,40 @@ def generate_probe(p: Panel, options: Options) -> str:
 
 	s += '''
 	drm_panel_add(&ctx->panel);
+'''
+
+	if p.has_dsc:
+		s += f'''
+	/* This panel only supports DSC; unconditionally enable it */
+	dsi->dsc = &ctx->dsc;
+
+	ctx->dsc.dsc_version_major = {p.dsc_version};
+	ctx->dsc.dsc_version_minor = {p.dsc_scr_version};
+
+	// TODO: Pass slice_per_pkt = {p.dsc_slice_per_pkt}
+	ctx->dsc.slice_height = {p.dsc_slice_height};
+	ctx->dsc.slice_width = {p.dsc_slice_width};
+	/*
+	 * hdisplay should be read from the selected mode once
+	 * it is passed back to drm_panel (in prepare?)
```

Comment:
Why is there a question mark here? Is this something someone generating the panel driver should fix or consider? I must admit I don't fully understand the meaning of this comment :')

Reply:
This is something upstream that needs to be fixed (and I discussed that a while ago with @lumag).

DRM bridges provide the mode (from Kernel Mode Setting), but the `drm_panel` shell doesn't provide it back to panel drivers.

Hence, if one of my panels define **multiple modes with different resolutions** (e.g. we have a 2.5k and 4k mode on most high-end 21:9 @60/120Hz Sony phones), we have **no way of knowing** which mode userspace ended up selecting, despite this information being **mandatory** for setting the right DSC configuration (and typically some DSI-DCS parameters).

`(in prepare?)` was referring to a guess that it should be passed in prepare, because during probe (where this comment resides) userspace couldn't have set a mode yet (chicken-egg problem). Hence all of this code should likely move to `.prepare` at some point.
Set-up work for that _might_ have already been done when DSC code was prepped for DisplayPort hotplugging, which requires a very similar thing (e.g. parameters can change at runtime and won't be fixated at probe once for the entire runtime of the kernel).